server/aws-lsp-codewhisperer/src/language-server/utils.ts

```diff
@@ -131,3 +133,15 @@ export const getUserAgent = (initializeParams: InitializeParams, serverInfo?: Se
 
     return items.join(' ')
 }
+
+export function getLoginTypeFromProvider(credentialsProvider: CredentialsProvider): LoginType {
+    if (credentialsProvider.hasCredentials('iam')) {
```

Comment:
Unfortunately, we cant rely on presence of `iam` type to decide which login type - `CredentialProvider allows both types to exist simultaneously, which might result in bug when client does not clean not used type correctly.

Instead, I think we should rely on the specific CodewhispererService implementation class used in server.